### PR TITLE
[ubuntu] Refactor PowerShell build scripts

### PR DESCRIPTION
**Pull request by shamil-mubarakshin**
# Description
This PR refactors PowerShell scripts used for Ubuntu builds:
1) `Get-ToolsetContent` helper function is used instead of direct commands
2) `Invoke-DownloadWithRetry` helper function is added
3) Local functions are renamed
4) Simplified conditions
5) Minor style changes

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/ubuntu/scripts/build/Install-PowerShellAzModules.ps1 [hunk 1]
```diff
@@ -28,17 +27,16 @@ foreach ($module in $modules)
...
+            | Select-Object -ExpandProperty files `
+            | Select-Object -First 1
+
+        if (-not $asset) {
+            Write-Host "Asset for ${moduleName} ${toolVersion} was not found in versions manifest"
             exit 1
         }
+
+        Write-Host "Downloading asset for ${moduleName} ${toolVersion}: $($asset.filename)"
+        wget $asset.download_url -nv --retry-connrefused --tries=10 -P $installPSModulePath
```
Comment (Alexey-Ayupov) [inline images/ubuntu/scripts/build/Install-PowerShellAzModules.ps1:39]: Should we use PowerShell native here instead of `wget`? E.g. `Invoke-RestMethod`

**Review by Alexey-Ayupov — DISMISSED** [review]

File: images/ubuntu/scripts/build/Configure-Toolset.ps1 [hunk 2]
```diff
@@ -4,91 +4,79 @@
...
     return $installationDir
 }
 
-function Add-EnvironmentVariable
-{
-    param
-    (
-        [Parameter(Mandatory)] [string] $Name,
-        [Parameter(Mandatory)] [string] $Value,
+function Add-ETCEnvironmentVariable {
```
Comment (mikhailkoliada) [inline images/ubuntu/scripts/build/Configure-Toolset.ps1:27]: I would not pin the function name to etcSomething, etc is just a path, I'd call it `Add-GlobalEnvironmentVariable`

**Review by Alexey-Ayupov — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**